client/utils/guidance.py

```diff
@@ -136,42 +144,107 @@ def show_guidance_msg(self, unique_id, input_lines, access_token, hash_key, guid
 
         # No list of misunderstandings for this wrong answer, default message
         if not lst_mis_u:
             log.info("Cannot find the list of misunderstandings.")
-            print(GUIDANCE_DEFAULT_MSG)
-            return EMPTY_MISUCOUNT_TGID_PRNTEDMSG
 
-        self.misU_count_dict = self.update_misUcounts(hash_key, lst_mis_u,
-                                                      repr(input_lines), shorten_unique_id)
+        wa_count_threshold = self.guidance_json['wrongAnsThresh']
+        wa_lst_assess_num = assess_dict_info['dictWA2LstAssessNum_WA']
+        msg_id_set = set()
 
-        Wrong_Answer_threshold = self.guidance_json['wrongAnsThresh']
+        answerDict, countData = self.get_misUdata()
+        prev_responses = answerDict.get(shorten_unique_id, [])
 
-        msg_id_set = set()
+        # Confirm that this WA has not been given before
+        seen_before = response in prev_responses
 
-        # If the count is higher than the threshold, we need to display the
-        # message
-        for mis_u in self.misU_count_dict:
-            if self.misU_count_dict[mis_u] >= Wrong_Answer_threshold:
-                # Add each associated misunderstanding ID to the set
-                msg_id = lambda_info_misu(dict_info, mis_u)
+        answerDict[shorten_unique_id] = prev_responses + [response]
+        self.save_misUdata(countData, answerDict)
 
-                if msg_id is not None:
-                    msg_id_set.add(msg_id)
+        if not seen_before:
+            # Lookup the list of assessNum and WA related to this wrong answer
+            # in the question's dictWA2LstAssessNum_WA
+            lst_assess_num = wa_lst_assess_num.get(response, [])
+            if not lst_assess_num:
+                log.info("Cannot get the lst of assess nums given this reponse.")
 
-        if self.tg_id == TG_CONTROL:
-            # if student is in control group, just print the default message
-            log.info("Student is in control group.")
-            print(GUIDANCE_DEFAULT_MSG)
-            return (self.misU_count_dict, self.tg_id, "")
+            # Check in answerDict to see if the student has ever given
+            # any of these wrong answers (sourced from dictWA2LstAssessNum_WA)
+            has_given_resp = any(other_resp in prev_responses for
+                                 _, other_resp in lst_assess_num)
+
+            if not has_given_resp:
+                log.info("Student has not given a response in dictWA2LstAssessNum_WA.")
+                print(GUIDANCE_DEFAULT_MSG)
+                return EMPTY_MISUCOUNT_TGID_PRNTEDMSG
+
+            # Check if the current wrong answer is in the question's dictWA2DictInfo
+            if not wa_details:
+                # Increment countDict by the number of wrong answers seen
+                # for each tag assoicated with this wrong answer
+                for misu in lst_mis_u:
+                    countData[misu] += min(len(prev_responses), 1)
```

Comment:
This should be incremented by the number of responses the student has given before that is also in the `lst_assess_num`.